<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Collatz Conjecture Calculator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>761</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>26</pointsize>
      <weight>75</weight>
      <italic>true</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="text">
     <string>Collatz Conjecture</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>381</width>
      <height>491</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>12</y>
       <width>351</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
       <weight>75</weight>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>DESCRIPTION:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>52</y>
       <width>351</width>
       <height>441</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;justify&quot;&gt;The &lt;span style=&quot; font-weight:600;&quot;&gt;Collatz Conjecture &lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;(also known as &lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;3n+1&lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;)&lt;/span&gt; is a famous problem in Mathematics.&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;It works on all positive integers and applies one of two rules to the given number. If the given integer is &lt;span style=&quot; font-weight:600;&quot;&gt;even&lt;/span&gt;, it is divided by 2. If, however, the integer is &lt;span style=&quot; font-weight:600;&quot;&gt;odd&lt;/span&gt;, it is multiplied by 3 and then summed with 1.&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;All integers &lt;span style=&quot; font-style:italic;&quot;&gt;(even and odd)&lt;/span&gt; eventually come down to a loop which goes on indefinitely. That loop is the &lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;4&lt;/span&gt;, &lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;2&lt;/span&gt;, &lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;1&lt;/span&gt; loop. That is, all numbers end up in this loop eventually.&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;The problem is that we do not know if all positive integers fall into this loop. The hope is to one day find a number that either goes onto infinity or makes another loop that is not the 4, 2, 1 loop.&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;This program is a demonstration of the problem for you people to interact with.&lt;/span&gt;&lt;/p&gt;&lt;p align=&quot;justify&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Hope you enjoy and hope this example brings you closer to understanding the complexities of Mathematics!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>409</x>
      <y>109</y>
      <width>371</width>
      <height>121</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>171</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
       <italic>false</italic>
      </font>
     </property>
     <property name="text">
      <string>Enter a number here:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="txt_box">
     <property name="geometry">
      <rect>
       <x>182</x>
       <y>20</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>1000000000</number>
     </property>
     <property name="placeholderText">
      <string>Only +ve integers allowed</string>
     </property>
    </widget>
    <widget class="QPushButton" name="button">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>70</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Calculate!</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>240</y>
      <width>371</width>
      <height>361</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>26</x>
       <y>22</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
       <weight>75</weight>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>OUTPUT:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="button_graph">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>20</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Plot Graph!</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="out_text">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>341</width>
       <height>291</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string>Collatz Conjecture output will show here</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <tabstops>
  <tabstop>txt_box</tabstop>
  <tabstop>button</tabstop>
  <tabstop>button_graph</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>txt_box</sender>
   <signal>returnPressed()</signal>
   <receiver>button</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>728</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
